### PR TITLE
WIP: Respond to distinct clang-tidy warings and errors on rtp.c [nomerge]

### PR DESCRIPTION
**Pull request by xrucka**
Pull request for changes introduced by resolving clang-tidy issues. 

Individual changes shall be commented on each commit.

DO NOT MERGE (yet)

### CODE REVIEW

File: src/rtp/rtp_callback.c [hunk 1]
```diff
@@ -233,7 +233,6 @@ void rtp_recv_callback(struct rtp *session, rtp_event * e)
                 process_sdes(participants, e->ssrc, (rtcp_sdes_item *) e->data);
                 break;
         case RX_APP:
-                pckt_app = (rtcp_app *) e->data;
```
Comment (xrucka) [inline src/rtp/rtp_callback.c:236]: This line is duplicit with the same line above, causing clang-tidy to complain about dead variables. Resolution: removed extra line.

File: src/rtp/rtp.c [hunk 2]
```diff
@@ -333,28 +332,25 @@ static inline int filter_event(struct rtp *session, uint32_t ssrc)
...
+                                cc++;
+                                continue;
                         }
+
+                        session->last_advertised_csrc++;
+                        if (session->last_advertised_csrc == session->csrc_count) {
+                                session->last_advertised_csrc = 0;
+                        }
+                        /* This returns each source marked "should_advertise_sdes" in turn. */
+                        return s->ssrc;
```
Comment (xrucka) [inline src/rtp/rtp.c:353]: There are two changes in this block:
1) C99-ization, removing the unneeded forward declarations.
2) Reformulating the inner loop to early-continue and thus move focus towards the real point of the function.

Resolution of clang-tidy's complaints on returns and elses.

File: src/rtp/rtp.c [hunk 3]
```diff
@@ -1593,106 +1586,106 @@ static int rtp_recv_data(struct rtp *session, uint32_t curr_rtp_ts)
 static void rtp_process_data(struct rtp *session, uint32_t curr_rtp_ts,
                uint8_t *buffer, rtp_packet *packet, int buflen)
 {
-        /* This routine preprocesses an incoming RTP packet, deciding whether to process it. */
-        uint8_t *buffer_vlen = NULL;
-        int vlen = 12;          /* vlen = 12 | 16 | 20 */
-        source *s;
+        if (buflen <= 0)
+	        return;
```
Comment (xrucka) [inline src/rtp/rtp.c:1591]: Branch swap of the original master-if. Most of this change is just whitespace change, as the whole branch code was unindented by one level.

File: src/rtp/rtp.c [hunk 4]
```diff
@@ -1593,106 +1586,106 @@ static int rtp_recv_data(struct rtp *session, uint32_t curr_rtp_ts)
...
-                                            ("RTP packet from probationary source ignored...\n");
-                                }
-                        } else {
-                                /* debug_msg("RTP packet from unknown source ignored\n"); */
-                        }
-                } else {
-                        session->invalid_rtp_count++;
-                        debug_msg("Invalid RTP packet discarded\n");
-                }
+        assert(packet != NULL);
```
Comment (xrucka) [inline src/rtp/rtp.c:1599]: Resolution of clang-tidy warning on numerous packet fileds null-dereference.

File: src/rtp/rtp.c [hunk 5]
```diff
@@ -1593,106 +1586,106 @@ static int rtp_recv_data(struct rtp *session, uint32_t curr_rtp_ts)
...
+		        if (s->probation == -1) {
+			        s->probation = MIN_SEQUENTIAL;
+			        s->max_seq = packet->seq - 1;
+		        }
+		        if (update_seq(s, packet->seq)) {
+			        process_rtp(session, curr_rtp_ts,
+				            packet, s);
+			        return; /* we don't free "packet", that's done by the callback function... */
+		        } 
+
```
Comment (xrucka) [inline src/rtp/rtp.c:1676]: Branch extracted -- resolution to clang-tidy's else-after-return

**Comment by mpiatka** [comment]
I've reviewed and merged some of the changes that were still applicable.

**Pull request closed without merging**